docs/reference.md

```diff
@@ -9219,6 +9243,7 @@ val value = DataTypes[Sc](
   myLocalDate = LocalDate.parse("2023-12-20"),
   myLocalTime = LocalTime.parse("10:15:30"),
   myLocalDateTime = LocalDateTime.parse("2011-12-03T10:15:30"),
+  myUtilDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS").parse("2011-12-03T10:15:30.000"),
```

Comment:
Should we use `SimpleDateFormat` here ?

 There are instances where `java.text.SimpleDateFormat` is used to parse dates. A better approach is to use `java.time.format.DateTimeFormatter` from the modern Java Time API (java.time) introduced in Java 8, which is thread-safe, more flexible, and offers better support for various date/time formats. 

We could also convert from LocalDateTime 
```scala
import java.time.format.DateTimeFormatter
import java.time.{LocalDateTime, ZoneId, ZoneOffset, ZonedDateTime}
import java.util.Date

val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
val date = Date.from(
  LocalDateTime.parse("2023-11-12 03:22:41", formatter)
    .toInstant(
      ZonedDateTime.now(ZoneId.systemDefault()).getOffset
    )
)

println(date)
```